src/Controls/src/BindingSourceGen/BindingSourceGenerator.cs

```diff
@@ -15,55 +15,186 @@ namespace Microsoft.Maui.Controls.BindingSourceGen;
-		});
+	static Binding? GetBindingForGeneration(GeneratorSyntaxContext context, CancellationToken t)
+	{
+		var invocation = (InvocationExpressionSyntax)context.Node;
+
+		var method = invocation.Expression as MemberAccessExpressionSyntax;
+
+
+		var sourceCodeLocation = new SourceCodeLocation(
+			context.Node.SyntaxTree.FilePath,
+			method!.Name.GetLocation().GetLineSpan().StartLinePosition.Line + 1,
+			method!.Name.GetLocation().GetLineSpan().StartLinePosition.Character + 1
+		);
+
+		var argumentList = invocation.ArgumentList.Arguments;
+		var getter = argumentList[1].Expression;
+
+
+		if (getter is not LambdaExpressionSyntax lambda)
+		{
+			return null; // TODO: Optimize
+		}
+
+
+		if (context.SemanticModel.GetSymbolInfo(lambda).Symbol is not IMethodSymbol symbol)
+		{
+			return null;
+		}; // TODO
+
+		var inputType = symbol.Parameters[0].Type;
+		var inputTypeGlobalPath = inputType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+
+		var outputType = symbol.ReturnType;
+		var outputTypeGlobalPath = symbol.ReturnType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+
+		var inputTypeIsGenericParameter = inputType.Kind == SymbolKind.TypeParameter;
+		var outputTypeIsGenericParameter = outputType.Kind == SymbolKind.TypeParameter;
+
+		var parts = new List<PathPart>();
+		var types = new List<string>();
+		ParsePath(lambda.Body, context, parts, types);
+
+
+		// Convert to string
+		var pathasString = string.Join("<>", parts.Select(p => p.MemberName));
+		var typesAsString = string.Join("<>", types);
+
+
+		return new Binding(
+			Id: ++_idCounter,
+			Location: sourceCodeLocation,
+			SourceType: new TypeName(inputTypeGlobalPath, IsNullable(inputType), inputTypeIsGenericParameter),
+			PropertyType: new TypeName(outputTypeGlobalPath, IsNullable(outputType), outputTypeIsGenericParameter),
+			Path: [.. parts],
+			GenerateSetter: true //TODO: Implement
+		);
 	}
 
-	private bool IsSetBindingMethod(SyntaxNode node)
+	static void ParsePath(CSharpSyntaxNode? expressionSyntax, GeneratorSyntaxContext context, List<PathPart> parts, List<string> types)
 	{
-		
+		if (expressionSyntax is null)
+		{
+			return;
+		}
+		if (expressionSyntax is IdentifierNameSyntax identifier)
+		{
+			types.Add("identifier");
+			var member = identifier.Identifier.Text;
+			var typeInfo = context.SemanticModel.GetTypeInfo(identifier).Type;
+			if (typeInfo == null)
+			{
+				return;
+			}; // TODO
+			var isNullable = IsNullable(typeInfo);
+			parts.Add(new PathPart(member, isNullable));
+			return;
+		}
+
+		if (expressionSyntax is MemberAccessExpressionSyntax memberAccess)
+		{
+			types.Add("memberAccess");
+			var member = memberAccess.Name.Identifier.Text;
+			var typeInfo = context.SemanticModel.GetTypeInfo(memberAccess.Expression).Type;
+			if (typeInfo == null)
+			{
+				return;
+			};
+			ParsePath(memberAccess.Expression, context, parts, types); //TODO: Nullable
```

Comment:
I wonder if we should use recursion. We could have just a `while (expressionSyntax is not null)` loop and just keep moving along the expression. We should keep performance in mind from the start :)

Reply:
I found out that the root of the expression is not necessericaly the first element. So if we have e.g. `foo?.property` the first thing we find is actually conditional access. Then in order to get `foo` we need to parse `conditionalAccess.Expression`. If this is the only edge case we can properly use simple while, if not we will have to choose between BFS or recursion. I will take a look into that.

Reply:
Yeah, I would expect that.

BTW the conditional access in the expression what we should use as indication of "is nullable". But getting the nullable annotations right will be definitely one of the "remaining 20%" problems.

Reply:
OK, let's keep it for now and we can refactor it later.